setuptools/config/_apply_pyprojecttoml.py

```diff
@@ -153,15 +153,17 @@ def _guess_content_type(file: str) -> Optional[str]:
     raise ValueError(f"Undefined content type for {file}, {msg}")
 
 
-def _long_description(dist: "Distribution", val: _DictOrStr, root_dir: StrPath):
+def _long_description(
+    dist: "Distribution", val: _LongDescriptionValue, root_dir: StrPath
+):
     from setuptools.config import expand
 
     if isinstance(val, str):
-        file: Union[str, list] = val
+        file: Union[str, tuple[()]] = val
         text = expand.read_files(file, root_dir)
         ctype = _guess_content_type(val)
     else:
-        file = val.get("file") or []
+        file = val.get("file") or ()
```

Comment:
In this case the runtime type of `val["file"]`[^1] is `Union[str, list[str]]`, because `tomlkit` maps TOML arrays into lists (and never produces tuples...).

Would it be better to define `file: Union[str, list[str]]` or `file: Union[str, Sequence[str]]`?

[^1]: when `"file" in val`

Reply:
The reason I used an empty tuple was to have a statically falsy value that matches an `Iterable[str]` for `expand.read_files`.

Atm it didn't end up changing anything as per the comment.

After looking at it again... isn't this actually showing that there's an issue?
`file` could be truthy with a list of files, in which case this would add a `list` to the `_referenced_files` set instead of a `str`

Reply:
Umm.. Good question.

I think I will need to have a look on this tomorrow morning.

I might be wrong about the whole `list[str]` here (but I need to double check, if this code regards the `project.readme` field in `pyproject.toml` and or `tool.setuptools.dynamic.readme`, if it is just the first it should only `str`, if it includes the second, then it is `str | List[str]`...)

Reply:
If it's the first one, then calling `expand.read_files` also seems overkill.

Documentation ref: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata

But I'll let you validate the code.

Reply:
So it seems to be only for `project.readme`, which means `str` if present.

In that case it should be fine to use `()`, in runtime the value will never be a `list[str]`.
I believe that `()` is better than `""` because I am using a comprehension/for-loop... In the comprehension, `()` would have no effect, but I don't know exactly what the effect of `""` would be. Will it try to open a file named `""`?

I used `read_files` because I did not want to re-implement the logic for checking for existing and local files.